bench/bench_skiplist.ml

```diff
@@ -0,0 +1,57 @@
+open Saturn
+
+let workload num_elems num_threads add remove =
+  let sl = Skiplist.create () in
+  let elems = Array.init num_elems (fun _ -> Random.int 10000) in
+  let push () =
+    Domain.spawn (fun () ->
+        let start_time = Unix.gettimeofday () in
+        for i = 0 to (num_elems - 1) / num_threads do
+          Domain.cpu_relax ();
+          let prob = Random.float 1.0 in
+          if prob < add then Skiplist.add sl (Random.int 10000) |> ignore
+          else if prob >= add && prob < add +. remove then
+            Skiplist.remove sl (Random.int 10000) |> ignore
+          else Skiplist.mem sl elems.(i) |> ignore
```

Comment:
The use of `elems` array here and above is a bit strange.  A huge array is initialized and then only the fraction for a single thread is used.  Why not use a `Random.int` here as well?